Marlin/src/pins/stm32f1/pins_LONGER3D_LK.h

```diff
@@ -99,10 +99,11 @@
 // Avoid nozzle heat and fan start before serial init
 #define BOARD_OPENDRAIN_MOSFETS
 
+// STM32 framework cannot use WRITE macros before HAL_init()
```

Comment:
STM32 HAL... not framework

Reply:
The framework will not work with IO done in global objects construtors (they run before system init).

but this fast io thing is stm32 specific.

anyway, if Marlin has a callback called HAL_Init, even if used only by one hal, it should be the first thing called and no IO should be done before it… :-/
The architecture should follow some logic

Reply:
Others HALs dont have this issue, its a weakness of STM32/ FastIO... and do a real crash of the board

Reply:
What is running before HAL_Init? That code cannot be changed to run after HAL_Init?